src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/EventCounter.cs

```diff
@@ -46,31 +46,53 @@ public EventCounter(string name, EventSource eventSource)
 
             InitializeBuffer();
             _name = name;
-            EventCounterGroup.AddEventCounter(eventSource, this);
+            _group = EventCounterGroup.GetEventCounterGroup(eventSource);
+            _group.Add(this);
+            _min = float.PositiveInfinity;
+            _max = float.NegativeInfinity;
         }
 
         /// <summary>
-        /// Writes the metric.
+        /// Writes 'value' to the stream of values tracked by the counter.  This updates the sum and other statistics that will 
+        /// be logged on the next timer interval.  
         /// </summary>
         /// <param name="value">The value.</param>
         public void WriteMetric(float value)
         {
             Enqueue(value);
         }
 
+#if EventCounterDispose
+        // TODO Go through API review and expose this.  
+        /// <summary>
+        /// Removes the counter from set that the EventSource will report on.
+        /// </summary>
+        public void Dispose()
+        {
+            _group.Remove(this);
+        }
```

Comment:
EventCounters are kept alive by their group which in turn is kept alive by the callback from the EventSources that the EventCounter is 'attached' to.    It works the other way as well, an EventCounter will keep the EventSource alive.   If they are both unreferenced, however they both will die (Neither requires Disposal).   

The explicit Dispose does detach the EventCounter early (I have also fixes this so that _group is nulled, and we check for null before using _group (this is the only place it is really used).